src/Worker/Core/IOrchestrationFilter.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.DurableTask.Worker;
+
+/// <summary>
+/// Defines a filter for validating orchestrations.
+/// </summary>
+public interface IOrchestrationFilter
```

Comment:
This needs `[Obsolete("Experimental")]`. Same for all new public APIs.

Reply:
Done.